code/contracts/ONEWallet.sol

```diff
@@ -64,30 +65,33 @@ contract ONEWallet {
         (uint32 ct, bool completed) = _findCommit(hash);
         require(ct == 0 && !completed, "Commit already exists");
         Commit memory nc = Commit(hash, uint32(block.timestamp), false);
-        commits.push(nc);
+        commits.push(nc); // IH: would not it be better to have mapping of commits?
```

Comment:
We discussed on Telegram. To include the discussion:

```
I had an early version of commit with map but quick scratched it. The issue is an attacker can commit a lot of garbage (at the expense of their own gas) and make the contract storage unnecessarily large, and unbounded. Thus, there needs to be some mechanisms for removing the map keys. Since we can’t iterate a map in solidity, we have to use an array to track it. Therefore, we have Commit[]. Nice catch with uint8 - forgot to fix that
```

You replied

```
For this cases is usefull EnumerableMap from openzeppelin library. Maybe you can check it
```

I checked EnumerableMap. It seems they rely on using an array to track the keys all the same. The contracts are quite heavy: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/EnumerableSet.sol and https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/structs/EnumerableMap.sol

So I think using a simple array to track and clean the commits are more gas-efficient than using EnumerableMap. Besides, in C++ and Java, looking up an element using linear search over a small array is actually much faster than using a map structure. I don't know if the same would apply to Solidity and EVM, but I suspect it is likely to behave the same.

Reply:
Okay, I will add a lookup map. I want to minimize the use of external dependencies at the core level, unless we have complex logics that really require these libraries. They make debugging and analysis harder, and uses more gas.